### PR TITLE
Add a TensorField

### PR DESCRIPTION
**Pull request by nelson-liu**
This PR adds a field for dealing with arbitrary Tensors in your dataset.

### CODE REVIEW

File: allennlp/tests/data/fields/tensor_field_test.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from numpy.testing import assert_allclose
+import pytest
+import torch
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.data.fields import TensorField
+
+
+class TestLabelField(AllenNlpTestCase):
```
Comment (joelgrus) [inline allennlp/tests/data/fields/tensor_field_test.py]: rename the class?

File: allennlp/data/fields/tensor_field.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    @overrides
+    def get_padding_lengths(self) -> Dict[str, int]:  # pylint: disable=no-self-use
+        return {}
+
+    @overrides
+    def as_tensor(self,
+                  padding_lengths: Dict[str, int],
+                  cuda_device: int = -1) -> torch.Tensor:
+        # pylint: disable=unused-argument,not-callable
+        return self.tensor if cuda_device == -1 else self.tensor.cuda(cuda_device)
```
Comment (joelgrus) [inline allennlp/data/fields/tensor_field.py]: technically this does the wrong thing if the original tensor is on the GPU and then you call it with cuda_device=-1. that's a pretty weird case, but probably we should handle it correctly.

I believe under 0.4 you could do something like

```
device = torch.device(f"cuda:{cuda_device}" if cuda_device >= 0 else "cpu")
return self.tensor.to(device)
```

File: allennlp/tests/data/fields/tensor_field_test.py (hunk 1 above)
Comment (nelson-liu) [reply allennlp/tests/data/fields/tensor_field_test.py]: whoops yes, sorry.

File: allennlp/data/fields/tensor_field.py (hunk 2 above)
Comment (nelson-liu) [reply allennlp/data/fields/tensor_field.py]: should this change be made for all the other fields as well?

Comment (joelgrus) [reply allennlp/data/fields/tensor_field.py]: no, because they're not starting with existing tensors

Comment (nelson-liu) [reply allennlp/data/fields/tensor_field.py]: right, thanks.

**Comment by nelson-liu** [comment]
@joelgrus : @DeNeutoy and I had an off-github conversation, and we decided that it might make the most sense to just slightly rework the `ArrayField` to work with torch Tensor input. does this sound reasonable to you?

**Comment by joelgrus** [comment]
I'm cool either way

**Pull request closed without merging**